app/src/main/java/org/sil/storyproducer/model/StoryIO.kt

```diff
@@ -218,6 +216,109 @@ fun isZipped(fileName: String?): Boolean {
         arrayOf("zip", "bloom", "bloomd", "bloomSource").contains(it)
     } == true
 }
+// copy a file to the new location
+fun copySubFile(context: Context, subFile: String, newWorkspaceFolder: DocumentFile, oldWorkspaceFolder: DocumentFile): Boolean {
+
+    val oldDoc = oldWorkspaceFolder.findFile(subFile) ?: return false
+    if (!oldDoc.isFile)
+        return false
+
+    val newDoc = newWorkspaceFolder.createFile(oldDoc.type!!, subFile) ?: return false
+
+    val buffer = ByteArray(1024 * 64)
+    val outStream = context.contentResolver.openOutputStream(newDoc.uri) ?: return false
+    val inStream = context.contentResolver.openInputStream(oldDoc.uri) ?: return false
+    var bytesRead = 0
+    do {
+        bytesRead = inStream.read(buffer)
+        if (bytesRead > 0)
+            outStream.write(buffer, 0, bytesRead)
+    } while (bytesRead > 0)
+    inStream.close()
+    outStream.close()
+
+    return true
+}
+
+// Copy a sub-folder and content to a new folder location - this is done recursively
+fun copySubFolder(context: Context, subFolder: String, newWorkspaceFolder: DocumentFile, oldWorkspaceFolder: DocumentFile): Boolean {
+
+    val oldFolder = oldWorkspaceFolder.findFile(subFolder) ?: return false
+    if (!oldFolder.isDirectory)
+        return false
+
+    val newFolder = newWorkspaceFolder.findFile(subFolder) ?:
+                        newWorkspaceFolder.createDirectory(subFolder) ?:
+                        return false
+
+    for (oldSubDoc in oldFolder.listFiles()) {
+        if (oldSubDoc.isDirectory) {
+            if (!copySubFolder(context, oldSubDoc.name!!, newFolder, oldFolder))
+                return false
+        } else if (oldSubDoc.isFile) {
+            if (!copySubFile(context, oldSubDoc.name!!, newFolder, oldFolder))
+                return false
+        } else
+            return false
+    }
+
+    return true
+}
+
+// Copy a sub-folder from the source templates folder and if ok return for further processing
+fun copyOldStory(context: Context, file: DocumentFile, newWorkspaceFolder: DocumentFile, oldWorkspaceFolder: DocumentFile): DocumentFile? {
+
+    do {
+        // NB: This list may also contains Stories and zipped Stories already found in the new SP Templates folder
+
+        if (newWorkspaceFolder.uri == oldWorkspaceFolder.uri)
+            return file // same folder so no need to copy
+
+        if (oldWorkspaceFolder.name?.isEmpty() ?: return file)
+            return file // no old folder so noting to copy
+
+        if (newWorkspaceFolder.name?.isEmpty() ?: break)
+            break   // no new SP Templates folder so can't copy
+
+        if (file.name?.isEmpty() ?: break)
+            break;  // no Story name so can't copy
+
+        if (file.name!! != WORD_LINKS_DIR &&
```

Comment:
in copysubfile function, we need to check that any copied wordlinks file does not already exist.